dag/src/job.rs

```diff
@@ -1,6 +1,46 @@
+use crate::{config::Config, input::Input};
+use plonky2::{field::extension::Extendable, hash::hash_types::RichField};
 
+pub struct Invocation<F: RichField + Extendable<D>, const D: usize> {
+    function_hash: [F; 4],
+    inputs: Vec<Input<F, D>>,
+}
 
-pub struct Job {
-    invocation: Invocation,
-    inputs: Vec<Inputs>,
-}
\ No newline at end of file
+impl<F: RichField + Extendable<D>, const D: usize> Invocation<F, D> {
+    pub fn new(function_hash: [F; 4], inputs: Vec<Input<F, D>>) -> Self {
+        Self {
+            function_hash,
+            inputs,
+        }
+    }
+}
+
+pub struct Job<F: RichField + Extendable<D>, const D: usize> {
+    invocation: Invocation<F, D>,
+    config: Config,
+}
+
+impl<F: RichField + Extendable<D>, const D: usize> Job<F, D> {
+    pub fn new(invocation: Invocation<F, D>, config: Config) -> Self {
+        Self { invocation, config }
+    }
+}
+
+pub enum Effect {
+    Error,
+    FutureEffect,
+    ContinueExecutation,
+}
+
+pub struct ExecuteResult<F: RichField + Extendable<D>, const D: usize> {
+    invocation_hash: [F; 4],
+    pure_output_hash: [F; 4],
```

Comment:
the output is an IPLD object, which can contain the outputs, possibly expressed as field elements; but I think these need to be objects that are IPLD seriablizable

Reply:
Refactored to use `[u8; 32]` instead